src/devices/wifi/nm-iwd-manager.c

```diff
@@ -544,9 +544,12 @@ interface_added (GDBusObjectManager *object_manager, GDBusObject *object,
 		id = known_network_id_new (name, security);
 
 		data = g_hash_table_lookup (priv->known_networks, id);
-		if (data)
+		if (data) {
+			_LOGW ("interface-added event for a KnownNetwork already present in "
+			       "priv->known_networks ('%s', %s)", name, type_str);
```

Comment:
Ok, changing this.  This was more indended as an assert for something that should never happen but should nevertheless be handled because we can't trust messages from DBus.

Reply:
Thank you. Just to elaborate on the reasoning:

 There are only
  - "real" assertions (any of `g_return_*()`, `g_warn*()`, `g_assert*()`, `nm_assert*()`, `assert()`). This is only for must-not-happen and bug-in-our-code scenarios
  - and regular runtime errors.

The latter requires error handling for the user (like a well crafted error message).